### PR TITLE
Adding the bulk of the feature

### PR DESCRIPTION
**Pull request by srtfisher**
Extracts the feature from Alleypack to be used in open-source packages.

### CODE REVIEW

File: src/class-block-converter.php [hunk 1]
```diff
@@ -0,0 +1,387 @@
...
+	/**
+	 * Create ul blocks.
+	 *
+	 * @param DOMNode $node The node.
+	 * @return Block
+	 */
+	protected function ul( DOMNode $node ): Block {
+		return new Block(
+			block_name: 'list',
+			content: static::get_node_html( $node ),
```
Comment (mogmarsh) [inline src/class-block-converter.php:181]: WordPress 6.1 introduces a list item block
https://github.com/WordPress/gutenberg/pull/42711

File: src/class-block-converter.php [hunk 2]
```diff
@@ -0,0 +1,387 @@
...
+	 * @param DOMNode $node The node.
+	 * @return block
+	 */
+	protected function ol( DOMNode $node ): Block {
+		return new Block(
+			block_name: 'list',
+			attributes: [
+				'ordered' => true,
+			],
+			content: static::get_node_html( $node ),
```
Comment (mogmarsh) [inline src/class-block-converter.php:227]: same on the list item blocks https://github.com/WordPress/gutenberg/pull/42711

File: src/class-block-converter.php [hunk 3]
```diff
@@ -0,0 +1,387 @@
...
+
+	/**
+	 * Removing whitespace between blocks
+	 *
+	 * @param string $block Gutenberg blocks.
+	 * @return string
+	 */
+	protected function minify_block( $block ) {
+		if ( preg_match( '/(\s){2,}/s', $block ) === 1 ) {
+			return preg_replace( '/(\s){2,}/s', '', $block );
```
Comment (mogmarsh) [inline src/class-block-converter.php:302]: do we want to replace with `''` or `' '`?

File: src/class-block-converter.php [hunk 4]
```diff
@@ -0,0 +1,387 @@
...
+'<!-- wp:paragraph -->
+<p><br></p>
+<!-- /wp:paragraph -->',
+'<!-- wp:html -->
+<div> </div>
+<!-- /wp:html -->',
+'<!-- wp:heading {"level":3} -->
+<h3>
+                                                        </h3>
+<!-- /wp:heading -->',
```
Comment (mogmarsh) [inline src/class-block-converter.php:368]: I feel like there should be a better way to do this, but good enough for now...

**Review by mogmarsh — APPROVED** [review]
a few little things. Overall 👍

File: src/class-block-converter.php (hunk 1 above)
Comment (srtfisher) [reply src/class-block-converter.php:181]: #2

File: src/class-block-converter.php (hunk 4 above)
Comment (srtfisher) [reply src/class-block-converter.php:368]: Yeah... I didn't write this part 🙉

**Pull request merged**